### ISSUE TITLE
Allow custom breakpoints to be defined

### ISSUE THREAD

**Issue by codepuncher** [beaverbuilder/feature-requests#139]
**Is your feature request related to a problem? Please describe.**
We use Bootstrap to build most of our frontend and gives our designers much more control and freedom with their designs. Clients often have differences in their list of most popular screen resolutions used by their users. One client may need designs to prioritise the 1366px width breakpoint. But that is one of the multiple breakpoints that a design will need to address.
Unfortunately, Beaver Builder only gives you 3 defined breakpoints to work with (small, medium and large). Whilst you can of course write your own CSS, it feels like a "hack" and not something that should need to be done. Why define some values in the BB settings and the rest in custom CSS? It can become difficult or tedious to manage.

**Describe the solution you'd like**
It would be a **huge improvement** for Beaver Builder to allow a custom set of breakpoints.
Maybe adding a PHP filter like `fl_builder_breakpoints` which would contain the current ones we see in the UI. We could then use the filter to add our own to the array.
```php
add_filter('fl_builder_breakpoints', function (array $breakpoints): array {
	$breakpoints['xxl'] = (object) [
		'label' => __('Extra Large Desktop', 'fl-builder'),
		'value' => 1600,
		'unit' => 'px',
	];

	return $breakpoints;
});
```
This is my preference and I imagine that I could also extract Bootstrap SCSS variables into JSON which I could then also use in the above filter to automatically add new breakpoints in a Bootstrap project.

**Describe alternatives you've considered**
Of course, you can just add your own CSS breakpoints (either in the BB Global CSS, WP Customiser or your own theme styles).
My current approach is adding a Bootstrap `.container` class to the `fl-col-group` element (e.g. `class="fl-col-group fl-node-606482b0a905e container"`). This allows our project-wide container max-widths and padding values to be applied to the rows. Although "containers", "rows" and "columns" don't work the same in BB, the solution works for some cases.

**Additional context**
None

**Comment by itineris-milos** [comment]
👍

**Comment by NikoNice44** [comment]
This 🙌🏻🙌🏻🙌🏻

**Comment by ramshengale** [comment]
👍

**Comment by csirvatka** [comment]
Yes!!!! I was glad to find this. I would LOVE **to be able to add more breakpoints!** There are always more than 2 necessary to have a responsive design look good: consider when the browser window is resized on a desktop or laptop. I always end up having to CSS in several media queries. I feel like BB relies heavily on the user having to resort to a LOT of CSS in general. This one thing would help save a lot of time and hair pulling for me!

**Comment by codente** [comment]
Just wanted to update that we did look into this for 2.6 but opted to go a different route and instead we'll be looking at adding another breakpoint.

**Comment by edellingham** [comment]
Just having an additional couple break points available would be super useful.

**Comment by csirvatka** [comment]
One additional would be good! Two would be even better! :-) Thank you though!! You could even add more set breakpoints like you have, but allow users to change the widths like we can now. Just a thought. Thank you!!

**Comment by codente** [comment]
> Just wanted to update that we did look into this for 2.6 but it would be very, very complex to support custom breakpoints. As such, we opted to go a different route and instead we'll be looking at adding another breakpoint.

Just a note on this.. we haven't ruled out custom breakpoints which is why this is still open.  :)

**Comment by jodzeee** [comment]
Thanks for the update!

**Comment by csirvatka** [comment]
Well that helps! Thank you!

**Comment by viktarbel** [comment]
Hi there, just checking in — would love to hear any updates on this. It would be extremely useful to have at least one additional breakpoint (and not have the largest one limited). We've been missing this feature for years, and it's still not implemented, which is disappointing and often complicates responsive adjustments.

P.S. Where can I officially raise an issue for the plugin?

**Comment by codente** [comment]
This is where you'd raise the feature request so it's already done. 😄   There's no update on this as it's not on our roadmap at this moment but that could change in the future.

Custom breakpoint support still has not been ruled out though which is why this is still open!

**Comment by viktarbel** [comment]
> This is where you'd raise the feature request so it's already done. 😄 There's no update on this as it's not on our roadmap at this moment but that could change in the future.
> 
> Custom breakpoint support still has not been ruled out though which is why this is still open!

Thank you, but I meant raising an issue, not suggesting a feature 😄 It's too sad that it's not even on your roadmap.

**Comment by codente** [comment]
You can see what IS on our roadmap here:
https://www.wpbeaverbuilder.com/beaver-builder-dev-update-the-road-to-2-10-and-beyond/

We do plan to keep doing these dev update posts so you can see what we're working on so keep an eye on the blog but if something changes with custom breakpoints making it onto the roadmap, I will update the issue and comment at that time.

**Comment by viktarbel** [comment]
Thank you, but where can I raise an issue request eventually 😄? I mean something that doesn’t work as it should.

**Comment by jodzeee** [comment]
Hoping this will be on the roadmap for the next major release!

**Comment by codente** [comment]
@jodzeee it is not!
